sentence_transformers/sparse_encoder/models/IDF.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import TYPE_CHECKING, Any
+
+import torch
+from safetensors.torch import load_file as load_safetensors_file
+from safetensors.torch import save_model as save_safetensors_model
+from transformers import AutoTokenizer
+
+if TYPE_CHECKING:
+    from transformers import PreTrainedTokenizer
+
+
+class IDF(torch.nn.Module):
+    def __init__(
+        self,
+        weight: torch.Tensor,
+        frozen: bool = False,
+        tokenizer: PreTrainedTokenizer | None = None,
+    ):
+        super().__init__()
+        self.weight = torch.nn.Parameter(weight, requires_grad=not frozen)
+        self.frozen = frozen
+        self.word_embedding_dimension = self.weight.size(0)
+        self.tokenizer = tokenizer
+        self.config_keys = ["frozen", "max_seq_length"]
+        self.max_seq_length = self.tokenizer.model_max_length
+
+    def forward(self, features: dict[str, torch.Tensor]) -> dict[str, torch.Tensor]:
+        input_ids = features["input_ids"]
+        batch_size = input_ids.shape[0]
+
+        multi_hot = torch.zeros(
+            batch_size, self.word_embedding_dimension, dtype=torch.float32, device=input_ids.device
+        )
+        batch_indices = torch.arange(batch_size, device=input_ids.device).unsqueeze(-1)
+        multi_hot[batch_indices, input_ids] = 1
+
+        sentence_embedding = multi_hot * self.weight
+
+        features["sentence_embedding"] = sentence_embedding
+        return features
+
+    def save(self, output_path, safe_serialization: bool = True) -> None:
+        """Save both the IDF model and its tokenizer if available"""
+        os.makedirs(output_path, exist_ok=True)
+
+        # Save the model weights
+        if safe_serialization:
+            save_safetensors_model(self, os.path.join(output_path, "model.safetensors"))
+        else:
+            torch.save(self.state_dict(), os.path.join(output_path, "pytorch_model.bin"))
+
+        # Save config with tokenizer info
+        config_dict = self.get_config_dict()
+
+        # Save tokenizer
+        self.tokenizer.save_pretrained(output_path)
+
+        with open(os.path.join(output_path, "config.json"), "w") as fOut:
+            json.dump(config_dict, fOut)
+
+    @classmethod
+    def from_json(cls, json_path: str, tokenizer, **config):
```

Comment:
```suggestion
    def from_json(cls, json_path: str, tokenizer: PreTrainedTokenizer, **config):
```
I missed this one earlier